### PR TITLE
Add improvements for default permission handling

### PR DESCRIPTION
**Pull request by NKavindya**
> To achieve this, I refactored the method to accept parameters for different permission sources, enabling it to handle lifecycle-state-related permissions and other default scopes dynamically. The default scopes, previously added through scattered sources, are now consolidated and fetched directly from the cdm configuration. For lifecycle-state-related permissions, I updated the application-mgt XML parsing logic to use the same reusable method, ensuring consistency.

> Additionally, I cleaned up redundant code by removing older implementations for default scope addition. The updated logic now centralizes default permission handling, improving maintainability. These changes make the permission management flow easier to extend in the future.

### CODE REVIEW

File: components/apimgt-extensions/io.entgra.device.mgt.core.apimgt.application.extension.api/src/main/webapp/META-INF/permissions.xml [hunk 1]
```diff
@@ -26,7 +26,6 @@
     it will result 403 error at the runtime.
 -->
 <PermissionConfiguration>
```
Comment (lasanthaDLPDS) [inline components/apimgt-extensions/io.entgra.device.mgt.core.apimgt.application.extension.api/src/main/webapp/META-INF/permissions.xml:28]: Can't we delete the whole file?

File: components/apimgt-extensions/io.entgra.device.mgt.core.apimgt.webapp.publisher/src/main/java/io/entgra/device/mgt/core/apimgt/webapp/publisher/APIPublisherServiceImpl.java [hunk 2]
```diff
@@ -460,14 +458,18 @@ public void addDefaultScopesIfNotExist() throws APIManagerPublisherException {
...
-                        List<String> bindings = new ArrayList<>(
-                                Arrays.asList(scopeMapping.getDefaultRoles().split(",")));
-                        bindings.add(ADMIN_ROLE_KEY);
+                        boolean isAssignableToDefaultRoles = defaultPermission.isAssignableToDefaultRoles() == null
+                                || defaultPermission.isAssignableToDefaultRoles();
+                        List<String> bindings = new ArrayList<>();
+                        if (isAssignableToDefaultRoles) {
+                            bindings.addAll(Arrays.asList(scopeMapping.getDefaultRoles().split(",")));
+                            bindings.add(ADMIN_ROLE_KEY);
+                        }
```
Comment (lasanthaDLPDS) [inline components/apimgt-extensions/io.entgra.device.mgt.core.apimgt.webapp.publisher/src/main/java/io/entgra/device/mgt/core/apimgt/webapp/publisher/APIPublisherServiceImpl.java:472]: In the catch block, don't catch chain of exceptions from a single catch block.

File: components/apimgt-extensions/io.entgra.device.mgt.core.apimgt.webapp.publisher/src/main/java/io/entgra/device/mgt/core/apimgt/webapp/publisher/APIPublisherStartupHandler.java [hunk 3]
```diff
@@ -103,16 +105,20 @@ public void completedServerStartup() {
             }
 
             try {
+                DeviceManagementConfig deviceManagementConfig = DeviceConfigurationManager.getInstance().getDeviceManagementConfig();
```
Comment (lasanthaDLPDS) [inline components/apimgt-extensions/io.entgra.device.mgt.core.apimgt.webapp.publisher/src/main/java/io/entgra/device/mgt/core/apimgt/webapp/publisher/APIPublisherStartupHandler.java:108]: Do we need to load the "DeviceManagementConfig" twice in this method?

File: components/apimgt-extensions/io.entgra.device.mgt.core.apimgt.webapp.publisher/src/main/java/io/entgra/device/mgt/core/apimgt/webapp/publisher/APIPublisherStartupHandler.java [hunk 4]
```diff
@@ -200,6 +204,9 @@ private void updateScopeMetadataEntryWithDefaultScopes() {
             log.error("Metadata entry already exists for " + Constants.PERM_SCOPE_MAPPING_META_KEY);
         } catch (MetadataManagementException e) {
             log.error("Error encountered while updating permission scope mapping metadata with default scopes");
+        } catch (PermissionManagementException e) {
+            String msg = "Error when adding default permission to the registry ";
+            log.error(msg, e);
```
Comment (lasanthaDLPDS) [inline components/apimgt-extensions/io.entgra.device.mgt.core.apimgt.webapp.publisher/src/main/java/io/entgra/device/mgt/core/apimgt/webapp/publisher/APIPublisherStartupHandler.java:209]: If you are not throwing an error, what's the point of having a String object here? There is no point of having a object, just log the message.

            log.error("Error when adding default permission to the registry ",  e);

Further, in the "MetadataManagementException" log the exception also.

File: components/application-mgt/io.entgra.device.mgt.core.application.mgt.common/pom.xml [hunk 5]
```diff
@@ -57,6 +57,8 @@
                             io.entgra.device.mgt.core.device.mgt.common.app.mgt;version="${io.entgra.device.mgt.core.version.range}",
                             io.entgra.device.mgt.core.device.mgt.common.operation.mgt;version="${io.entgra.device.mgt.core.version.range}",
                             io.entgra.device.mgt.core.device.mgt.core.common.exception;version="${io.entgra.device.mgt.core.version.range}",
+                            io.entgra.device.mgt.core.device.mgt.core.config.*,
```
Comment (lasanthaDLPDS) [inline components/application-mgt/io.entgra.device.mgt.core.application.mgt.common/pom.xml:60]: Add the version.

File: components/application-mgt/io.entgra.device.mgt.core.application.mgt.core/pom.xml [hunk 6]
```diff
@@ -52,6 +52,8 @@
                             io.entgra.device.mgt.core.apimgt.application.extension;version="${io.entgra.device.mgt.core.version.range}",
                             io.entgra.device.mgt.core.apimgt.application.extension.dto;version="${io.entgra.device.mgt.core.version.range}",
                             io.entgra.device.mgt.core.apimgt.application.extension.exception;version="${io.entgra.device.mgt.core.version.range}",
+                            io.entgra.device.mgt.core.apimgt.webapp.publisher.*,
+                            io.entgra.device.mgt.core.apimgt.webapp.publisher.exception*,
```
Comment (lasanthaDLPDS) [inline components/application-mgt/io.entgra.device.mgt.core.application.mgt.core/pom.xml:56]: Add the version

File: components/application-mgt/io.entgra.device.mgt.core.application.mgt.core/pom.xml [hunk 7]
```diff
@@ -440,6 +445,26 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.entgra.device.mgt.core</groupId>
+            <artifactId>io.entgra.device.mgt.core.apimgt.webapp.publisher</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-lang3</artifactId>
+            <version>3.12.0</version>
```
Comment (lasanthaDLPDS) [inline components/application-mgt/io.entgra.device.mgt.core.application.mgt.core/pom.xml:465]: Don't specify the version number in pom files of sub-modules.

File: components/application-mgt/io.entgra.device.mgt.core.application.mgt.core/pom.xml [hunk 8]
```diff
@@ -440,6 +445,26 @@
...
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon.apimgt</groupId>
+            <artifactId>org.wso2.carbon.apimgt.impl</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.entgra.device.mgt.core</groupId>
+            <artifactId>io.entgra.device.mgt.core.apimgt.webapp.publisher</artifactId>
```
Comment (lasanthaDLPDS) [inline components/application-mgt/io.entgra.device.mgt.core.application.mgt.core/pom.xml:460]: Add the "provided" scope

File: components/application-mgt/io.entgra.device.mgt.core.application.mgt.core/pom.xml [hunk 9]
```diff
@@ -440,6 +445,26 @@
...
         </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon.apimgt</groupId>
+            <artifactId>org.wso2.carbon.apimgt.api</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon.apimgt</groupId>
+            <artifactId>org.wso2.carbon.apimgt.impl</artifactId>
+            <scope>test</scope>
```
Comment (lasanthaDLPDS) [inline components/application-mgt/io.entgra.device.mgt.core.application.mgt.core/pom.xml:456]: Whats the reason for having test scope here?

File: components/certificate-mgt/io.entgra.device.mgt.core.certificate.mgt.api/src/main/webapp/META-INF/permissions.xml [hunk 10]
```diff
@@ -28,7 +28,6 @@
     it will result 403 error at the runtime.
 -->
 <PermissionConfiguration>
-    <APIVersion></APIVersion>
```
Comment (lasanthaDLPDS) [inline components/certificate-mgt/io.entgra.device.mgt.core.certificate.mgt.api/src/main/webapp/META-INF/permissions.xml:31]: We can delete the whole file right?

File: components/certificate-mgt/io.entgra.device.mgt.core.certificate.mgt.cert.admin.api/src/main/webapp/META-INF/permissions.xml [hunk 11]
```diff
@@ -28,7 +28,6 @@
     it will result 403 error at the runtime.
 -->
 <PermissionConfiguration>
-    <APIVersion></APIVersion>
```
Comment (lasanthaDLPDS) [inline components/certificate-mgt/io.entgra.device.mgt.core.certificate.mgt.cert.admin.api/src/main/webapp/META-INF/permissions.xml:31]: We can delete the whole file, right?

File: components/device-mgt/io.entgra.device.mgt.core.device.mgt.core/src/main/java/io/entgra/device/mgt/core/device/mgt/core/authorization/DeviceAccessAuthorizationServiceImpl.java [hunk 12]
```diff
@@ -234,11 +230,4 @@ private String removeTenantDomain(String username) {
     private int getTenantId() {
         return CarbonContext.getThreadLocalCarbonContext().getTenantId();
     }
-
-    private boolean addAdminPermissionToRegistry() throws PermissionManagementException {
-        Permission permission = new Permission();
-        permission.setName(CDM_ADMIN);
```
Comment (lasanthaDLPDS) [inline components/device-mgt/io.entgra.device.mgt.core.device.mgt.core/src/main/java/io/entgra/device/mgt/core/device/mgt/core/authorization/DeviceAccessAuthorizationServiceImpl.java:240]: Do we need to keep "CDM_ADMIN" constant.

File: components/device-mgt/io.entgra.device.mgt.core.device.mgt.core/src/main/java/io/entgra/device/mgt/core/device/mgt/core/authorization/GroupAccessAuthorizationServiceImpl.java [hunk 13]
```diff
@@ -165,12 +161,4 @@ private String removeTenantDomain(String username) {
     private int getTenantId() {
         return CarbonContext.getThreadLocalCarbonContext().getTenantId();
     }
-
-    private boolean addAdminPermissionToRegistry() throws PermissionManagementException {
-        Permission permission = new Permission();
-        permission.setName(GROUP_ADMIN);
```
Comment (lasanthaDLPDS) [inline components/device-mgt/io.entgra.device.mgt.core.device.mgt.core/src/main/java/io/entgra/device/mgt/core/device/mgt/core/authorization/GroupAccessAuthorizationServiceImpl.java:171]: Do we need to keep "GROUP_ADMIN" constant?

**Review by lasanthaDLPDS — CHANGES_REQUESTED** [review]

File: components/apimgt-extensions/io.entgra.device.mgt.core.apimgt.webapp.publisher/pom.xml [hunk 14]
```diff
@@ -176,6 +176,7 @@
                             io.entgra.device.mgt.core.device.mgt.common.metadata.mgt,
                             io.entgra.device.mgt.core.device.mgt.core.config,
                             io.entgra.device.mgt.core.device.mgt.core.config.permission,
+                            io.entgra.device.mgt.core.device.mgt.core.permission.mgt.*;version="${io.entgra.device.mgt.core.version.range}",
```
Comment (R4JxF60) [inline components/apimgt-extensions/io.entgra.device.mgt.core.apimgt.webapp.publisher/pom.xml:179]: Imports the required package, no need to import all the packages which match the wildcard.

File: components/application-mgt/io.entgra.device.mgt.core.application.mgt.common/pom.xml (hunk 5 above)
Comment (R4JxF60) [reply components/application-mgt/io.entgra.device.mgt.core.application.mgt.common/pom.xml:60]: Import the required packages, remove the wildcards.

**Pull request merged**